jmix-flowui/flowui/src/main/java/io/jmix/flowui/sys/XmlInheritanceProcessor.java

```diff
@@ -0,0 +1,316 @@
+            if (index < 0 || index > elements.size()) {
+                String message = String.format(
+                        "Incorrect extension XML for screen. Could not paste new element %s to position %s",
+                        newElement.getName(), index);
+
+                throw new DevelopmentException(message,
+                        ParamsMap.of("element", newElement.getName(), "index", index));
+            }
+            elements.add(index, newElement);
+        }
+        justAdded.add(newElement);
+        process(newElement, element);
+    }
+
+    protected interface ElementTargetLocator {
+        boolean suitableFor(Element extElem);
+        @Nullable
+        Element locate(Element resultParentElem, Element extElem);
+    }
+
+    protected static class CommonElementTargetLocator implements ElementTargetLocator {
+
+        @Override
+        public boolean suitableFor(Element extElem) {
+            return !StringUtils.isBlank(extElem.attributeValue("id"));
+        }
+
+        @Nullable
+        @Override
+        public Element locate(Element resultParentElem, Element extElem) {
+            String id = extElem.attributeValue("id");
+            for (Element e : resultParentElem.elements()) {
+                if (id.equals(e.attributeValue("id"))) {
+                    return e;
+                }
+            }
+            return null;
+        }
+    }
+
+    protected static class ViewElementTargetLocator implements ElementTargetLocator {
+
+        @Override
+        public boolean suitableFor(Element extElem) {
+            return "view".equals(extElem.getName());
+        }
+
+        @Nullable
+        @Override
+        public Element locate(Element resultParentElem, Element extElem) {
+            String entity = extElem.attributeValue("entity");
+            String clazz = extElem.attributeValue("class");
+            String name = extElem.attributeValue("name");
+            for (Element e : resultParentElem.elements()) {
+                if (name != null) {
+                    if (name.equals(e.attributeValue("name"))
+                            && ((entity != null && entity.equals(e.attributeValue("entity")))
+                            || (clazz != null && clazz.equals(e.attributeValue("class"))))) {
+                        return e;
+                    }
+                }
+            }
+
+            Element extParentElem = extElem.getParent();
+            if (extParentElem != null && Objects.equals(resultParentElem.attributeValue("id"),
+                    extParentElem.attributeValue("id"))) {
+                for (Element e : resultParentElem.elements()) {
+                    if ("view".equals(e.getName())) {
+                        return e;
+                    }
+                }
+            }
+
+            return null;
+        }
+    }
+
+    protected static class ViewPropertyElementTargetLocator implements ElementTargetLocator {
```

Comment:
This `ElementTargetLocator` handles nested into `view` element  (which is `fetchPlan` in CUBA terms) properties.

1. Requires refactoring and testing if it can be applied to the `fetchPlan` nested elements.
2. Requires renaming as it clashes with view (aka screen), e.g. `FeetchPlanPropertyElementTargetLocator`

Reply:
I adapted FetchPlanElementTargetLocator and FetchPlanPropertyElementTargetLocator and added a test for FetchPlanPropertyElementTargetLocator.